pkg/informer/informer.go

```diff
@@ -54,32 +57,30 @@ type DynamicDiscoverySharedInformerFactory struct {
 	workspaceLister tenancylisters.ClusterWorkspaceLister
 	disco           clusterDiscovery
 	dynamicClient   dynamic.Interface
-	handlers        []GVREventHandler
 	filterFunc      func(interface{}) bool
 	pollInterval    time.Duration
 	indexers        cache.Indexers
 
-	mu            sync.RWMutex // guards gvrs
-	gvrs          map[schema.GroupVersionResource]struct{}
-	informers     map[schema.GroupVersionResource]informers.GenericInformer
-	informerStops map[schema.GroupVersionResource]chan struct{}
-	terminating   bool
-}
+	handlersLock sync.Mutex
+	handlers     atomic.Value
```

Comment:
a lock plus atomic? This looks very odd now. I could see either or.

Reply:
Otherwise you risk data loss with 2 simultaneous writers